util/encoding.go

```diff
@@ -0,0 +1,165 @@
+package util
+
+import (
+	"encoding/base64"
+	"fmt"
+)
+
+var (
+	fibEncodeNumOutOfRangeErr = fmt.Errorf("the number to be encoded is out of range")
+)
+
+// enlarge enlarges the underlying byte slice.
+func (bs *BitStream) enlarge(n uint16) {
+	final := (bs.p + n + 7) / 8
+	for final > bs.Len() {
```

Comment:
It would probably by more efficient to append one mass of bytes, rather than looping over an append statement like this. 
`
size := final - bs.Len()
chuck := make([]byte, size, size)
bs.b = append(bs.b, chunk...)
`
The new array created by make should contain all zeros in all the elements of the array. The documentation isn't 100% clear, but that is how it reads to me.

This may not be a big difference when enlarge is used by the write functions below, as it will only be writing one or two bytes at a time. So I am not too worried about the final form of enlarge currently, but may be more of a factor if we make more expansive write functions in the future. In particular `WriteIntRange()` could probably calculate the total size needed fairly quickly, and then ensure the byte slice in the bit stream has sufficient capacity in one step, rather than depending on a sequence of `enlarge()` calls.